src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,164 @@
+
+    /**
+     * Class constructor of Storage.
+     * Initialize by adding a zeroth element so that the index of tasks can start from 1.
+     */
+    public TaskList() {
+        this.list.add(new ToDo("zeroth"));
+    }
+
+    /**
+     * Add a Todo Task to the TaskList.
+     * @param input the content of the new Todo Task
+     * @param needPrint the indicator that shows whether the message needs to be printed
+     * @throws DukeException if the content is empty
+     */
+    public void add(String input, boolean needPrint) throws DukeException {
+        if (isEmpty(input)) {
+            throw new DukeException("The description of a todo cannot be empty.");
+        }
+        ToDo newTask = new ToDo(input);
+        this.list.add(newTask);
+        if (needPrint) {
+            System.out.println("Got it. I've added this task:");
+            System.out.println(newTask.toString());
+            System.out.println("Now you have " + (this.list.size() - 1) + " tasks in the list.");
+        }
+    }
+
+    /**
+     * Add a Deadline Task to the TaskList.
+     * @param input the content of the new Deadline Task
+     * @param deadline the deadline date and time of the new Deadline Task
+     * @param needPrint the indicator that shows whether the message needs to be printed
+     * @throws DukeException if the content is empty
+     */
+    public void add(String input, String deadline, boolean needPrint) throws DukeException {
+        if (isEmpty(input) || isEmpty(deadline)) {
+            throw new DukeException("The description and deadline of a deadline cannot be empty.");
+        }
+        Deadline newTask = new Deadline(input, deadline);
+        this.list.add(newTask);
+        if (needPrint) {
+            System.out.println("Got it. I've added this task:");
+            System.out.println(newTask.toString());
+            System.out.println("Now you have " + (this.list.size() - 1) + " tasks in the list.");
+        }
+    }
+
+    /**
+     * Add a Event Task to the TaskList.
+     * @param input the content of the new Event Task
+     * @param from the start date and time of the new Event Task
+     * @param to the end date and time of the new Event Task
+     * @param needPrint the indicator that shows whether the message needs to be printed
+     * @throws DukeException if the content is empty
+     */
+    public void add(String input, String from, String to, boolean needPrint) throws DukeException {
+        if (input.equals("") || from.equals("") || to.equals("")) {
+            throw new DukeException("The description and duration of an event cannot be empty.");
+        }
+        Event newTask = new Event(input, from, to);
+        this.list.add(newTask);
+        if (needPrint) {
+            System.out.println("Got it. I've added this task:");
+            System.out.println(newTask.toString());
+            System.out.println("Now you have " + (this.list.size() - 1) + " tasks in the list.");
+        }
+    }
```

Comment:
Perhaps making the function names more descriptive in terms of which tasks are being added would be better?